src/components/internal/InternalCheckbox.tsx

```diff
@@ -6,27 +6,36 @@ import useBackground from "../../hooks/useBackground";
 import useResponsivePropsCSS from "../../hooks/useResponsivePropsCSS";
 import VisuallyHidden from "../VisuallyHidden";
 
-const COLORS = ["grey.t05", "white"];
+const COLORS = ["grey.t05", "white"] as const;
 
 const DEFAULT_PROPS = {
   color: "grey.t05",
   disabled: false,
   __internal__keyboardFocus: false,
-};
+} as const;
 
 InternalCheckbox.COLORS = COLORS;
 InternalCheckbox.DEFAULT_PROPS = DEFAULT_PROPS;
 
-function CheckboxIcon({ color, isChecked }) {
+type CheckboxColor = "grey.t05" | "white";
+
+interface CheckboxIconProps {
+  color: CheckboxColor;
+  isChecked?: boolean;
+}
+
+function CheckboxIcon({ color, isChecked = false }: CheckboxIconProps) {
   const theme = useTheme();
   const { inputColorMap } = useBackground();
   const rectCSS = useResponsivePropsCSS(
     {},
     {},
     {
+      // @ts-ignore
```

Comment:
How about adding types to this bit ?

Reply:
have a look at the `useResponsiveCSS` code and you'll understand why. It's extremely complex and I want to do it but it will require alot of time.